### PR TITLE
Handle concurrent write conflict

### LINKED ISSUE #1195: Set repeatable read isolation level
To mitigate issues caused by concurrent requests, we should set isolation level = repeatable read for transactions for now. If update failed, we can respond with HTTP Conflict status code and let client decide whether to retry request.

### PR DESCRIPTION
**Pull request by kiootic**
fix #1195

### CODE REVIEW

File: pkg/core/db/sql_executor.go [hunk 1]
```diff
@@ -75,7 +82,18 @@ func (e *SQLExecutor) QueryRowWith(sqlizeri sq.Sqlizer) (*sqlx.Row, error) {
...
+		}
 		return nil, errors.WithDetails(err, errors.Details{"sql": errors.SafeDetail.Value(sql)})
 	}
 	return db.QueryRowxContext(e.context, sql, args...), nil
 }
+
+func isWriteConflict(err error) bool {
+	var pqErr *pq.Error
+	if errors.As(err, &pqErr) {
+		return pqErr.Code == "40001" || pqErr.Code == "40P01"
```
Comment (louischan-oursky) [inline pkg/core/db/sql_executor.go]: Just curious. How did you find out these are the error codes we want to detect?

File: pkg/core/middleware/recover.go [hunk 2]
```diff
@@ -33,10 +34,15 @@ func (m RecoverMiddleware) Handle(next http.Handler) http.Handler {
...
-				handled := false
+				const errorTypeUnexpected = 0
+				const errorTypeHandled = 1
+				const errorTypeConflict = 2
+				errorType := errorTypeUnexpected
 				if herr, ok := err.(handler.HandledError); ok {
-					handled = true
+					errorType = errorTypeHandled
 					err = herr.Error
+				} else if err == db.ErrWriteConflict {
```
Comment (louischan-oursky) [inline pkg/core/middleware/recover.go]: errors.Is

File: pkg/core/db/sql_executor.go (hunk 1 above)
Comment (kiootic) [reply pkg/core/db/sql_executor.go]: These are the error codes corresponded to the two errors mentioned in [documentation page on isolation level](https://www.postgresql.org/docs/9.5/transaction-iso.html).

Comment (louischan-oursky) [reply pkg/core/db/sql_executor.go]: 40P01 is not mentioned in that page. Where can I find it?

Comment (kiootic) [reply pkg/core/db/sql_executor.go]: You can find the error codes in https://www.postgresql.org/docs/9.1/errcodes-appendix.html. 40P01 is deadlock error, which is may indicate conflicts between running transactions.

Comment (louischan-oursky) [reply pkg/core/db/sql_executor.go]: Can you include these in the code as comments?

**Pull request merged**